### ISSUE TITLE
Zero configuration graphql constructor for spaces with format

### ISSUE THREAD

**Issue by Totktonada** [tarantool/graphql.0#60]
No options — no connections. Ability to provide just connections, but any other metainfo to be retrieved from tarantool.

Related (but not the blocker): https://github.com/tarantool/tarantool/issues/3080

**Comment by Totktonada** [comment]
I think we should enable such automatic configurators as generators of existing format of avro-schema, collections, sevice fields, indexes and so on to better decompose things to components.

**Comment by SudoBobo** [comment]
I suggest to use the following constructor format:
```
tarantool_graphql.new(cfg)
```
where
cfg = { auto_cfg, schemas, collections, service_fields,  indexes, accessor } - with the same meaning as earlier, execept `accessor` - now it may contain an accessor (as earlier) or a string with desired accessor type ('space' or 'shard' ) and auto_cfg - true or false - either auto generation should be done or not

And the following mechanism of construction:

1. if auto_cfg is `false` then we use cfg as earlier. 
2. If auto_cfg is `true` then we produce cfg from tarantool metadata and try to merge it with passed cfg. There are two purposes of this merging:
    1. To complete cfg with fields that can't be currently produced from tarantool metadata 
    (connections and nested elements of schema)
    2. To overwrite auto generated config with user written

**Comment by Totktonada** [comment]
Consider the following usage:

```
tarantool> graphql = require('graphql') -- no new()
tarantool> graphql.cfg{...}
tarantool> graphql.execute('query ...')
```

**Comment by SudoBobo** [comment]
@Totktonada Please check this mechanism of config generation.

1. Generate cfg.schemas from space:format()
2. Generate cfg.collections with no connections and names == schemas.names
3. Generate cfg.indexes ({ collection_name = indexes } map ) using collections from step 2
4. Merge generated config with user-written config (in case of overlapping, user-written config has higher prio) In his config user may provide additional fields for generated schemas (e.g. avro record or map fields)
5. Complement cfg. Complement simplified user-written connections to standard connections
6. Check that in resulting cfg all mentioned collections (in cfg.indexes, cfg.service_fields, etc) exist and. Similar checks on schemas

Also, what about service_fields? Can they be taken from tarantool?

**Comment by Totktonada** [comment]
Okay.

4. Note: merge arrays with named items per items by its names. Use field = box.NULL to delete an item. Configuration merger can be separate module.

> Also, what about service_fields? Can they be taken from tarantool?

Nope, it always is empty for space format determined objects.

**Comment by Totktonada** [comment]
@SudoBobo Can you please check whether this issue can be closed?

**Comment by SudoBobo** [comment]
Closed with #95 .
